pkg/faketikv/cluster.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package faketikv
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/pd/server/core"
+)
+
+// ClusterInfo records all cluster information
+type ClusterInfo struct {
+	*core.RegionsInfo
+	Nodes       map[uint64]*Node
+	firstRegion *core.RegionInfo
+}
+
+// GetBootstrapInfo returns first region and it's leader store
+func (c *ClusterInfo) GetBootstrapInfo() (*metapb.Store, *metapb.Region) {
+	storeID := c.firstRegion.Leader.GetStoreId()
+	store := c.Nodes[storeID]
+	return store.Store, c.firstRegion.Region
+}
+
+func (c *ClusterInfo) nodeHealth(storeID uint64) bool {
+	n, ok := c.Nodes[storeID]
+	if !ok {
+		return false
+	}
+	return n.Store.GetState() == metapb.StoreState_Up
+}
+
+func (c *ClusterInfo) stepLeader(region *core.RegionInfo) {
+	if c.nodeHealth(region.Leader.GetStoreId()) {
```

Comment:
No corresponding `stepFollower`? If a region's leader store is down, how can it elect next leader?